cmake --build build fails on SFML (no member named 'auto_ptr' in namespace 'std' and other errors)

```

/Users/lukasz/Desktop/projects/2021/tspart/deps/SFML/src/SFML/Audio/AudioDevice.cpp:128:10: error: no member named 'auto_ptr' in namespace 'std'
    std::auto_ptr<AudioDevice> device;
    ~~~~~^
/Users/lukasz/Desktop/projects/2021/tspart/deps/SFML/src/SFML/Audio/AudioDevice.cpp:128:19: error: 'AudioDevice' does not refer to a value
    std::auto_ptr<AudioDevice> device;
                  ^
/Users/lukasz/Desktop/projects/2021/tspart/deps/SFML/src/SFML/Audio/AudioDevice.hpp:46:7: note: declared here
class AudioDevice
      ^
/Users/lukasz/Desktop/projects/2021/tspart/deps/SFML/src/SFML/Audio/AudioDevice.cpp:128:32: error: use of undeclared identifier 'device'
    std::auto_ptr<AudioDevice> device;
                               ^
/Users/lukasz/Desktop/projects/2021/tspart/deps/SFML/src/SFML/Audio/AudioDevice.cpp:130:9: error: use of undeclared identifier 'device'
        device.reset(new AudioDevice);
        ^
```

![image](https://user-images.githubusercontent.com/12752520/140647875-de183c5a-55ce-4e80-a3a0-7fc1c7f8bd11.png)

--- a/deps/CMakeLists.txt
+++ b/deps/CMakeLists.txt
@@ -1,2 +1,4 @@
+set(CMAKE_CXX_STANDARD 14)
+
 add_subdirectory(SFML)
 add_subdirectory(TGUI)

